slasher/db/kv/spanner_new.go

```diff
@@ -70,14 +75,19 @@ func (db *Store) EpochSpans(ctx context.Context, epoch uint64) (*types.EpochStor
 }
 
 // SaveEpochSpans accepts a epoch and span byte array and writes it to disk.
-func (db *Store) SaveEpochSpans(ctx context.Context, epoch uint64, es *types.EpochStore) error {
+func (db *Store) SaveEpochSpans(ctx context.Context, epoch uint64, es *types.EpochStore, toCache bool) error {
 	ctx, span := trace.StartSpan(ctx, "slasherDB.SaveEpochSpans")
 	defer span.End()
 
 	if len(es.Bytes())%int(types.SpannerEncodedLength) != 0 {
 		return types.ErrWrongSize
 	}
 
+	if toCache {
```

Comment:
we should update cache if it exists for a certain epoch even if we are being asked to store to db in order to avoid having two diff values for the same epoch

Reply:
Thinking of this, if the function always writes to cache (if it exists) regardless of what its being told to, can't that cause problems for when we genuinely want to write to the DB? I think this is something we can control by being careful where we use what. 

If it always writes to cache if it has the key then the only way we can get a for sure DB write if something is evicted from the cache.

Reply:
i thought of this flow 
if tocache -> write to cache
if todb -> write to db -> check if epoch exists in cache -> update cache

Reply:
Thanks! Updated.